runtime/tests/interpreter/interpreter_test.go

```diff
@@ -7567,6 +7568,205 @@ func TestInterpretVariableDeclarationSecondValue(t *testing.T) {
+                let value: String
+
+                init() {
+                    self.value = "test"
+                }
+            }
+
+            resource R1 {
+                var r2: @R2?
+
+                init() {
+                    self.r2 <- nil
+                }
+
+                destroy() {
+                    destroy self.r2
+                }
+
+                fun moveToStack_Borrow_AndMoveBack(): &R2 {
+                    // The second assignment should not lead to the resource being cleared
+                    let optR2 <- self.r2 <- nil
+                    let r2 <- optR2!
+                    let ref = &r2 as &R2
+                    self.r2 <-! r2
+                    return ref
+                }
+            }
+
+            fun createR1(): @R1 {
+                return <- create R1()
+            }
+
+            fun test(r1: &R1): [String?] {
+                let r2 <- create R2()
+                r1.r2 <-! r2
+                let ref = r1.moveToStack_Borrow_AndMoveBack()
+                let value = r1.r2?.value
+                let refValue = ref.value
+                return [value, refValue]
+            }
+        `)
+
+		r1, err := inter.Invoke("createR1")
+		require.NoError(t, err)
+
+		r1 = r1.Transfer(inter, interpreter.EmptyLocationRange, atree.Address{1}, false, nil)
+
+		r1Type := checker.RequireGlobalType(t, inter.Program.Elaboration, "R1")
+
+		ref := &interpreter.EphemeralReferenceValue{
+			Value:        r1,
+			BorrowedType: r1Type,
+		}
+
+		value, err := inter.Invoke("test", ref)
+		require.NoError(t, err)
+
+		AssertValuesEqual(
+			t,
+			inter,
+			interpreter.NewArrayValue(
+				inter,
+				interpreter.EmptyLocationRange,
+				interpreter.VariableSizedStaticType{
+					Type: interpreter.OptionalStaticType{
+						Type: interpreter.PrimitiveStaticTypeString,
+					},
+				},
+				common.ZeroAddress,
+				interpreter.NewUnmeteredSomeValueNonCopying(
+					interpreter.NewUnmeteredStringValue("test"),
+				),
+				interpreter.NewUnmeteredSomeValueNonCopying(
+					interpreter.NewUnmeteredStringValue("test"),
+				),
+			),
+			value,
+		)
+
+		storage := inter.Storage()
+
+		var permanentSlabs []atree.Slab
+
+		for _, slab := range storage.(interpreter.InMemoryStorage).Slabs {
+			if slab.ID().Address == (atree.Address{}) {
+				continue
+			}
+
+			permanentSlabs = append(permanentSlabs, slab)
+		}
+
+		require.Equal(t, 2, len(permanentSlabs))
+
+		sort.Slice(permanentSlabs, func(i, j int) bool {
+			a := permanentSlabs[i].ID()
+			b := permanentSlabs[j].ID()
+			return a.Compare(b) < 0
+		})
+
+		var storedValues []string
+
+		for _, slab := range permanentSlabs {
+			storedValue := interpreter.StoredValue(inter, slab, storage)
+			storedValues = append(storedValues, storedValue.String())
+		}
+
+		require.Equal(t,
+			[]string{
+				`S.test.R1(r2: S.test.R2(value: "test", uuid: 2), uuid: 1)`,
+				`S.test.R2(value: "test", uuid: 2)`,
+			},
+			storedValues,
+		)
+	})
+}
+
```

Comment:
Fix incorrect conflict resolution: Remove interpreter tests, as they were moved to checker tests

```suggestion
```